docs/core-concepts/atomic-swap-htlc.md

```diff
@@ -0,0 +1,133 @@
+---
+id: atomic-swap-htlc
+title: Atomic Swaps using HTLCs
+sidebar_label: Atomic Swaps using HTLCs
+---
+
+[Atomic Swap](#atomic-swap)s using [HTLC](#htlc)s defines a solution using timelocks and hashlocks for exchanging two digital assets without the need of a trusted third party.
+
+Prior to the actual swap execution there is usually a [Negotiation](negotiation.md) phase where the two parties agree on the ledgers, assets and amounts to be swapped.
+
+Assuming that Alice has ETH on Ethereum and Bob has BTC on Bitcoin, Alice wants Bob's BTC and Bob wants Alice's ETH, the execution of a swap would be like this:
+
+1. Alice generates a secret and hashes it. She sends to Bob:
```

Comment:
I am assuming you are describing atomic swaps with HTLCs _in general_ here and not how exactly we are implementing them.

Re: "She sends to Bob":
You are mixing our current implementation (which to some extend is an implementation detail) with the general protocol of atomic swaps.

If we don't want this this documentation to be outdated, I would omit these kind of details here and explain atomic swaps using HTLCs without caring about the communication (you could say something like "they agree on identities, amounts and expiry times").

Reply:
I can see why you would raise this, and I would like to have more opinions on that. To some extent I want to be objective in this section and describe concepts outside of COMIT, however, I also feel we can put *some* influence of our implementation details into the core concept documentation. This way it might be easier for users to understand our solution.

That might be a bit far fetched though 😅 
@bonomat @D4nte @yosriady what is your opinion on that? Should I change this to be more general? (this would mean, that I would add another sentence before the slide showing the swap, to explain that this is how we do it in COMIT...)

Reply:
I tend to agree with @thomas here.  Keeping it more general makes it read more objective, which I think is a good thing to aim for.  Good work sticking at this @da-kami, writing good docs is tough.